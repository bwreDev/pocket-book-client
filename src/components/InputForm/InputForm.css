.input-button {
  background-color: #225a2a;
  color: white;
  border-radius: 15px;
  padding: 20px;
  border: none;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 10px;
  width: 250px;
}

.input-button:hover {
  background-color: lightgreen;
  color: #303030;
  cursor: pointer;
}

select {
  display: block;
  border: none;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.3;
  padding: 0.6em 1.4em 0.5em 0.8em;
  width: 75%;
  max-width: 75%;
  margin: 15px auto 15px auto;
  border-radius: 0.5em;
  background-color: lightgreen;
  cursor: pointer;
}

select option {
  font-weight: normal;
  background-color: #225a2a;
  color: white;
  cursor: pointer;
}

fieldset {
  border: none;
}

input {
  height: 40px;
  border-radius: 15px;
  border-style: none;
  margin: 5px;
  width: 75%;
  padding: 10px;
  background-color: lightgreen;
  cursor: pointer;
}

textarea {
  padding: 10px;
  height: 60px;
  border-radius: 15px;
  border-style: none;
  margin: 5px;
  width: 75%;
  font-size: 16px;
  background-color: lightgreen;
  cursor: pointer;
}

@media only screen and (min-width: 600px) {
  input {
    margin-top: 20px;
    margin-bottom: 20px;
    height: 60px;
    font-size: 20px;
  }

  select {
    font-size: 20px;
    font-weight: normal;
  }

  textarea {
    height: 80px;
    font-size: 20px;
  }
}

@media only screen and (min-width: 1200px) {
  input {
    margin-top: 30px;
    margin-bottom: 30px;
    height: 80px;
    font-size: 24px;
  }

  select {
    font-size: 24px;
    font-weight: normal;
  }

  textarea {
    height: 100px;
    font-size: 24px;
  }
}
