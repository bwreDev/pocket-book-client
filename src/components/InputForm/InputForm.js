import React, { Component } from 'react';
import InputsContext from '../../contexts/InputsContext';
import PocketBookApiService from '../../services/pocket-book-api-service';
import Navbar from '../../components/Navbar/Navbar';
import './InputForm.css';

export default class InputForm extends Component {
  static contextType = InputsContext;

  state = {
    input_type: null,
  };

  onChange = (input) => {
    this.setState({
      input_type: input.target.value,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();

    let eventObject = {
      title: this.state.input_type,
    };

    if (eventObject.title === 'income') {
      eventObject.content = event.target.description.value;

      eventObject.amount = event.target.amount.value;
    }
    if (eventObject.title === 'debt') {
      eventObject.content = event.target.description.value;

      eventObject.amount = event.target.amount.value;
    }

    PocketBookApiService.postInput(eventObject)
      .then(this.context.addInput)
      .then(this.props.history.push('/user'))
      .catch(this.context.setError);
  };

  render() {
    return (
      <>
        <Navbar />
        <h2>Please select income or debt.</h2>
        <form className='input-form' onSubmit={this.handleSubmit}>
          <h3>Type</h3>
          <select onChange={this.onChange} name='input_type'>
            <option value=''></option>
            <option value='income'>Incomes</option>
            <option value='debt'>Debts</option>
          </select>
          <fieldset className='amount'>
            <label htmlFor='amount'>
              <h3>Amount</h3>
            </label>
            <input placeholder='$0' type='number' name='amount' />
          </fieldset>
          <fieldset className='description'>
            <label htmlFor='description'>
              <h3>Description</h3>
            </label>
            <textarea
              placeholder='Describe your transaction'
              name='description'
            ></textarea>
          </fieldset>
          <button className='input-button' type='submit'>
            Save
          </button>
        </form>
      </>
    );
  }
}
