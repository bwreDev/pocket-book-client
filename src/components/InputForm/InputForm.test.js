import React from 'react';
import ReactDOM from 'react-dom';
import InputForm from './InputForm';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

it('should render without error', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <InputForm />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <InputForm />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
