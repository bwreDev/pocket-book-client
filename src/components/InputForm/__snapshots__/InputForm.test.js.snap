// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <nav
    className="navbar"
    role="navigation"
  >
    <ul
      className="navbar-list"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>,
  <h2>
    Please select income or debt.
  </h2>,
  <form
    className="input-form"
    onSubmit={[Function]}
  >
    <h3>
      Type
    </h3>
    <select
      name="input_type"
      onChange={[Function]}
    >
      <option
        value=""
      />
      <option
        value="income"
      >
        Incomes
      </option>
      <option
        value="debt"
      >
        Debts
      </option>
    </select>
    <fieldset
      className="amount"
    >
      <label
        htmlFor="amount"
      >
        <h3>
          Amount
        </h3>
      </label>
      <input
        name="amount"
        placeholder="$0"
        type="number"
      />
    </fieldset>
    <fieldset
      className="description"
    >
      <label
        htmlFor="description"
      >
        <h3>
          Description
        </h3>
      </label>
      <textarea
        name="description"
        placeholder="Describe your transaction"
      />
    </fieldset>
    <button
      className="input-button"
      type="submit"
    >
      Save
    </button>
  </form>,
]
`;
